integrationtests/tests/fixtures/mod.rs

```diff
@@ -64,128 +64,214 @@ use real::{RealBitcoinTest, RealLightningTest};
-    bitcoin::secp256k1::Secp256k1::new()
-}
+impl Fixtures {
+    pub fn new(num_peers: u16, amount_tiers: &[Amount]) -> Self {
+        let base_port = BASE_PORT.fetch_add(num_peers * 2 + 1, Ordering::Relaxed);
+
+        // in case we need to output logs using 'cargo test -- --nocapture'
+        if base_port == 4000 {
+            tracing_subscriber::fmt()
+                .with_env_filter(
+                    EnvFilter::try_from_default_env()
+                        .unwrap_or_else(|_| EnvFilter::new("info,fedimint::consensus=warn")),
+                )
+                .init();
+        }
 
-/// Generates the fixtures for an integration test and spawns API and HBBFT consensus threads for
-/// federation nodes starting at port 4000.
-pub async fn fixtures(
-    num_peers: u16,
-    amount_tiers: &[Amount],
-) -> (
-    FederationTest,
-    UserTest,
-    Box<dyn BitcoinTest>,
-    GatewayTest,
-    Box<dyn LightningTest>,
-) {
-    let base_port = BASE_PORT.fetch_add(num_peers * 2 + 1, Ordering::Relaxed);
-
-    // in case we need to output logs using 'cargo test -- --nocapture'
-    if base_port == 4000 {
-        tracing_subscriber::fmt()
-            .with_env_filter(
-                EnvFilter::try_from_default_env()
-                    .unwrap_or_else(|_| EnvFilter::new("info,fedimint::consensus=warn")),
-            )
-            .init();
-    }
+        let params = ServerConfigParams {
+            hbbft_base_port: base_port,
+            api_base_port: base_port + num_peers,
+            amount_tiers: amount_tiers.to_vec(),
+        };
+        let peers = (0..num_peers as u16).map(PeerId::from).collect::<Vec<_>>();
+
+        let max_evil = hbbft::util::max_faulty(peers.len());
+        let (server_config, client_config) =
+            ServerConfig::trusted_dealer_gen(&peers, max_evil, &params, OsRng::new().unwrap());
+
+        let disable_mocks = env::var("FM_TEST_DISABLE_MOCKS").is_ok();
+        let mut real_dir: Option<PathBuf> = None;
+        if disable_mocks {
+            let dir = env::var("FM_TEST_DIR").expect("Must have test dir defined for real tests");
+            real_dir = Some(PathBuf::from(dir));
+        }
 
-    let params = ServerConfigParams {
-        hbbft_base_port: base_port,
-        api_base_port: base_port + num_peers,
-        amount_tiers: amount_tiers.to_vec(),
-    };
-    let peers = (0..num_peers as u16).map(PeerId::from).collect::<Vec<_>>();
+        Self {
+            num_peers,
+            peers,
+            base_port,
+            server_config,
+            client_config,
+            disable_mocks,
+            real_dir,
+        }
+    }
 
-    let max_evil = hbbft::util::max_faulty(peers.len());
-    let (server_config, client_config) =
-        ServerConfig::trusted_dealer_gen(&peers, max_evil, &params, OsRng::new().unwrap());
+    // Builds (fed, user, bitcoin)
+    pub async fn build(&self) -> (FederationTest, UserTest, Box<dyn BitcoinTest>) {
```

Comment:
The name could indicate the difference better. `build_basic()` or something like that maybe?

Reply:
lets go with `make` `make_smth`